api/odigos/v1alpha1/instrumentationconfig_types.go

```diff
@@ -105,6 +100,10 @@ type InstrumentationLibraryConfig struct {
 	InstrumentationLibraryId InstrumentationLibraryId `json:"libraryId"`
 
 	TraceConfig *InstrumentationLibraryConfigTraces `json:"traceConfig,omitempty"`
+
+	// Configuration boolean options which are enabled for this instrumentation library.
+	// All options are disabled by default. To enable an option, add it's name to this list.
+	EnabledOptions []string `json:"enabledOptions,omitempty"`
```

Comment:
I think that a string alone for an option might be too specific, we can have specific options in the future which might have an integer value for them.
An option can have a string key and an *optional* string value - WDYT?

Reply:
I think in the future, we will have options with various types: numbers, strings, string arrays, etc.
Since they have different types, I want to have different properties in the CRD for each type.

For boolean options, I was thinking we only need to have them if they are `true` so the `false` value is the default.

That is why I choose to write it this way and postpone the other types for the future. 
But perhaps it's better to have the final version here so that we don't need to make other changes down the road

Reply:
made it better with support for multiple types